components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/signup/UserSelfSignUpManager.java

```diff
@@ -0,0 +1,298 @@
+     * the NotificationResponseBean.
+     * @throws IdentityRecoveryException If error occurs while user registration.
+     */
+    public NotificationResponseBean registerUser(UserBean userBean, String domainName, Property[] properties) throws
+            IdentityRecoveryException {
+
+        if (!config.isSelfSignUpEnabled()) {
+            throw Utils.handleClientException(ErrorCodes.DISABLE_SELF_SIGN_UP, null);
+        }
+
+        User user;
+
+        // Add user to self sign-up role.
+        try {
+
+            List<String> groupIds = new ArrayList<>();
+            groupIds.add(getSelfSignUpGroupId(domainName));
+            user = realmService.getIdentityStore().addUser(userBean, groupIds, domainName);
+        } catch (IdentityStoreException e) {
+            throw Utils.handleServerException(ErrorCodes.SELF_SIGN_UP_STORE_ERROR, null, e);
+        }
+
+        //TODO:Add User Object.
+        NotificationResponseBean notificationResponseBean = new NotificationResponseBean(user.getUniqueUserId());
+        notificationResponseBean.setUser(user);
+
+        IdentityMgtMessageContext messageContext = new IdentityMgtMessageContext();
+
+        Map<String, Object> props = new HashMap<>();
+        props.put(IdentityStoreConstants.USER, user);
+        props.put(SELF_SIGN_UP_PROPERTIES, properties);
+
+        Event event = new Event(SELF_SIGN_UP_EVENT, props);
+
+        try {
+
+            eventService.pushEvent(event, messageContext);
+
+            String confirmationCode = (String) messageContext.getParameter(CONFIRMATION_CODE);
+
+            // If notification internally managed, the confirmation code will be null.
+            notificationResponseBean.setCode(confirmationCode);
+        } catch (IdentityException e) {
+            throw Utils.handleServerException(ErrorCodes.TRIGGER_NOTIFICATION,
+                                              user.getUniqueUserId(), e);
+        }
+        return notificationResponseBean;
+    }
+
+    /**
+     * Check whether user is already confirmed or not.
+     *
+     * @param uniqueUserId Unique ID of the user.
+     * @return True if the user is already confirmed.
+     * @throws IdentityRecoveryException If error occurs while user registration.
+     */
+    public boolean isUserConfirmed(String uniqueUserId) throws IdentityRecoveryException {
+
+        if (StringUtils.isBlank(uniqueUserId)) {
+
+            throw new IdentityRecoveryClientException(ErrorCodes.INVALID_USER_ID.getCode(),
+                                                      ErrorCodes.INVALID_USER_ID.getMessage());
+        }
+
+        UserRecoveryDataStore userRecoveryDataStore = JDBCRecoveryDataStore.getInstance();
+        UserRecoveryData load = userRecoveryDataStore.loadByUserUniqueId(uniqueUserId);
+
+        boolean isUserConfirmed = false;
+
+        if (load == null || !RecoveryScenarios.SELF_SIGN_UP.equals(load.getRecoveryScenario())) {
```

Comment:
It is not good to only check NOT "SELF_SIGN_UP" as isUserConfirmed=true. Better to have a list of confirmed cases and check against all of them.